observation/src/test/java/com/linecorp/armeria/server/observation/MicrometerObservationServiceTest.java

```diff
@@ -0,0 +1,265 @@
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MicrometerObservationRegistryUtils;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.RpcRequest;
+import com.linecorp.armeria.common.RpcResponse;
+import com.linecorp.armeria.common.brave.HelloService;
+import com.linecorp.armeria.common.brave.SpanCollector;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestLogBuilder;
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.server.HttpService;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.TransientHttpService;
+import com.linecorp.armeria.server.TransientServiceOption;
+
+import brave.Span.Kind;
+import brave.Tracing;
+import brave.handler.MutableSpan;
+import brave.handler.SpanHandler;
+import brave.http.HttpTracing;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.CurrentTraceContext.ScopeDecorator;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.sampler.Sampler;
+
+class MicrometerObservationServiceTest {
+
+    private static final String TEST_SERVICE = "test-service";
+
+    private static final String TEST_METHOD = "hello";
+
+    @AfterEach
+    public void tearDown() {
+        Tracing.current().close();
+    }
+
+    @Test
+    @Disabled("TODO: We do it differently?")
+    void newDecorator_shouldFailFastWhenRequestContextCurrentTraceContextNotConfigured() {
+        assertThatThrownBy(() -> MicrometerObservationService.newDecorator(
+                MicrometerObservationRegistryUtils.observationRegistry(
+                        HttpTracing.create(Tracing.newBuilder().build()))))
+                .isInstanceOf(IllegalStateException.class).hasMessage(
+                        "Tracing.currentTraceContext is not a RequestContextCurrentTraceContext " +
+                        "scope. Please " +
+                        "call Tracing.Builder.currentTraceContext(RequestContextCurrentTraceContext" +
+                        ".ofDefault())."
+                );
+    }
```

Comment:
@jrhee17 Could you undisable and fix it?

Reply:
Let me remove this test 😅 

Essentially the original test was checking if `BraveClient` threw an exception if not used with `RequestContextCurrentTraceContext`.

In our current setup, we actually encourage usage of brave's default `ThreadLocalCurrentTraceContext` (although we could still use `RequestContextCurrentTraceContext`)

If we want to enable this, this would mean we need to migrate over this validation which I didn't think makes much sense. Let me know if you feel otherwise though.

https://github.com/line/armeria/blob/262fb4f0dd70a6c34b1c900b3a9156edc62cfcb1/brave/src/main/java/com/linecorp/armeria/internal/common/brave/TraceContextUtil.java#L62